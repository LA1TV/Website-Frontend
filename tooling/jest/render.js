import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const render = (Component, props = {}) => {
  return shallow(<Component {...props} />)
}

const renderToJson = (Component, props) => {
  return toJson(render(Component, props))
}

export { renderToJson, render }
