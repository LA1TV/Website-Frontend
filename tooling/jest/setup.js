let usersLa1tvApiKey

beforeAll(() => {
  usersLa1tvApiKey = process.env.LA1TV_API_KEY

  process.env.LA1TV_API_KEY = 'someApiKey'
})

beforeEach(() => {
  process.env.LA1TV_API_KEY = 'someApiKey'
})

afterAll(() => {
  process.env.LA1TV_API_KEY = usersLa1tvApiKey
})
