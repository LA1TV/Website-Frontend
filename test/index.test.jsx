import { renderToJson } from "tooling/jest/render";
import Index from "pages/index.js";

test("should be able to render", () => {
  expect(renderToJson(Index)).toMatchInlineSnapshot(`
    <Paragraph>
      Hello Next.js
    </Paragraph>
  `);
});
