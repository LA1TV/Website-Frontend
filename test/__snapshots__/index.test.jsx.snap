// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be able to render 1`] = `
<div>
  <p>
    Hello Next.js
  </p>
</div>
`;
